backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.flightleg
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.withStyle
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun BpkFlightLeg(
+  departureArrivalTime: String,
+  description: AnnotatedString,
+  stopsInfo: String,
+  duration: String,
+  modifier: Modifier = Modifier,
+  nextDayArrival: String? = null,
+  highlightStopsInfo: Boolean = false,
+  operatedBy: String? = null,
+  warning: String? = null,
+  contentDescription: String? = null,
+  carrierLogoContent: @Composable (BoxScope.() -> Unit)? = null,
+) {
+  Box(
+    modifier = modifier
+      .fillMaxWidth()
+      .semantics(mergeDescendants = true) {
+        if (contentDescription == null) {
+          // prevent screen readers from reading this allowing a broader
+          // content description to be set on the parent node
+          invisibleToUser()
+        } else {
+          this.contentDescription = contentDescription
+        }
+      },
+  ) {
+    Row(modifier = Modifier.fillMaxWidth()) {
+      carrierLogoContent?.let {
+        Box(content = it)
+        Spacer(modifier = Modifier.width(BpkSpacing.Base))
+      }
+      Column(
+        horizontalAlignment = Alignment.Start,
+      ) {
+        BpkText(
+          text = buildAnnotatedString {
+            append(departureArrivalTime)
+            nextDayArrival?.let {
+              withStyle(
+                SpanStyle(
+                  color = BpkTheme.colors.textPrimary,
+                  fontSize = BpkTheme.typography.label3.fontSize,
+                  baselineShift = BaselineShift.Superscript,
+                ),
+              ) {
+                append(it)
+              }
+            }
+          },
+          color = BpkTheme.colors.textPrimary,
+        )
+        mutableListOf<AnnotatedString>().apply {
```

Comment:
compose generally isn't very compatible with mutable types. rather than creating spans here could you just have separate `BpkText` composables? that way you can also apply the full text style, rather than just the font size (that might be something we need to figure out how we can solve actually for spans, but that seems outside of the scope of this PR)

Reply:
sure.
what do you mean by 
> apply the full text style, rather than just the font size
?
I'd have multiple BpkText with style `caption` and color `textSecondary`. Is that what you'd expect?

Reply:
yep that's what i mean. this is different to the span where we just apply the text size of the style due to limitations (which we may need to figure out how to solve within backpack, but that's on us)